src/init.cpp

```diff
@@ -1449,6 +1449,7 @@ bool AppInitMain()
                 // Note that it also sets fReindex based on the disk flag!
                 // From here on out fReindex and fReset mean something different!
                 if (!LoadBlockIndex(chainparams)) {
+                    if (ShutdownRequested()) break;
                     strLoadError = _("Error loading block database");
```

Comment:
I'll try to follow up a refactor here: this error should come `LoadBlockIndex` and therefore the above condition is not necessary.